scala/scala.bzl

```diff
@@ -21,360 +21,375 @@ _srcjar_filetype = FileType([".srcjar"])
-  javac_sources_cmd = ""
-  compile_java_srcs = len(java_srcs) != 0
-  if (compile_java_srcs):
-  # Set up the args to pass to javac because they can be too long for bash
-    javac_args_file = ctx.new_file(ctx.outputs.jar, ctx.label.name + "_javac_args")
-    javac_args = """{javac_opts} -classpath "{jars}:{out}_tmp" -d {out}_tmp {files}""".format(
-      javac_opts=" ".join(ctx.attr.javacopts),
-      jars=":".join([j.path for j in jars]),
-      files=" ".join([f.path for f in java_srcs]),
-      out=ctx.outputs.jar.path
-      )
-    ctx.file_action(output = javac_args_file, content = javac_args)
-    javac_sources_cmd = """
-    cat {javac_args} {{out}}_args/files_from_jar > {{out}}_args/java_args
-    {javac} {{jvm_flags}} @{{out}}_args/java_args""".format(javac_args = javac_args_file.path,javac=ctx.file._javac.path)
-
-  srcjar_cmd = ""
-  if len(all_srcjars) > 0:
-    srcjar_cmd = "\nmkdir -p {out}_tmp_expand_srcjars\n"
-    for srcjar in all_srcjars:
-      # Note: this is double escaped because we need to do one format call
-      # per each srcjar, but then we are going to include this in the bigger format
-      # call that is done to generate the full command
-
-      #TODO would like to be able to switch >/dev/null, -v, etc based on the user's settings
-      srcjar_cmd += """
-unzip -o {srcjar} -d {{out}}_tmp_expand_srcjars >/dev/null
-""".format(srcjar = srcjar.path)
-    srcjar_cmd += """find {out}_tmp_expand_srcjars -type f -name "*.scala" > {out}_args/files_from_jar\n"""
-
-  cmd = """
-rm -rf {out}_args
-rm -rf {out}_tmp
-rm -rf {out}_tmp_expand_srcjars
-set -e
-mkdir -p {out}_args
-touch {out}_args/files_from_jar
-mkdir -p {out}_tmp""" + srcjar_cmd + """
-cat {scalac_args} {out}_args/files_from_jar > {out}_args/scala_args
-env JAVACMD={java} {scalac} {jvm_flags} @{out}_args/scala_args""" + javac_sources_cmd + """
-# add any resources
-{cp_resources}
-{java} -jar {jar} -m {manifest} {out} {out}_tmp
-rm -rf {out}_args
-rm -rf {out}_tmp
-rm -rf {out}_tmp_expand_srcjars
-""" + ijar_cmd
-  cmd = cmd.format(
-      cp_resources=cp_resources,
-      java=ctx.file._java.path,
-      jvm_flags=" ".join(["-J" + flag for flag in ctx.attr.jvm_flags]),
-      scalac=ctx.file._scalac.path,
-      scalac_args=scalac_args_file.path,
-      out=ctx.outputs.jar.path,
-      manifest=ctx.outputs.manifest.path,
-      jar=_get_jar_path(ctx.files._jar),
-      ijar=ctx.file._ijar.path,
+    jars = _jars
+    ijar_output_path = ""
+    ijar_cmd_path = ""
+    if buildijar:
+        ijar_output_path = ctx.outputs.ijar.path
+        ijar_cmd_path = ctx.file._ijar.path
+
+    java_srcs = _java_filetype.filter(ctx.files.srcs)
+    sources = _scala_filetype.filter(ctx.files.srcs) + java_srcs
+    srcjars = _srcjar_filetype.filter(ctx.files.srcs)
+    all_srcjars = set(srcjars + list(dep_srcjars))
+    # look for any plugins:
+    plugins = _collect_plugin_paths(ctx.attr.plugins)
+    plugin_arg = ",".join(list(plugins))
```

Comment:
comma's are relatively common, so wouldn't surprise me if its in some one of their args :/. one option would be to json encode the list itself i suppose?